app/src/main/java/de/westnordost/streetcomplete/overlays/address/AddressOverlayForm.kt

```diff
@@ -169,10 +186,11 @@ class AddressOverlayForm : AbstractOverlayForm() {
 
         number?.streetHouseNumber?.let { lastHouseNumber = it }
         if (number is HouseAndBlockNumber) { number.blockNumber.let { lastBlockNumber = it } }
+            else lastBlockNumber = null
```

Comment:
```suggestion
        lastBlockNumber = if (number is HouseAndBlockNumber) number.blockNumber else null
```

Reply:
Is the `let` here really useless? It looks like, but I thought it's probably here for a reason...

Reply:
Introduced in a438117f7a85846ae0ab782d0695efb75cfd6360. It likely was a refactoring from a nullable `blockNumber`, where `?.let { … }` makes sense.